contracts/src/base/registry/facets/distribution/v2/RewardsDistributionStorage.sol

```diff
@@ -0,0 +1,26 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.18;
+
+// interfaces
+
+// libraries
+import {StakingRewards} from "./StakingRewards.sol";
+
+// contracts
+
+library RewardsDistributionStorage {
+  // keccak256(abi.encode(uint256(keccak256("facets.registry.rewards.distribution.v2.storage")) - 1)) & ~bytes32(uint256(0xff))
+  bytes32 internal constant STORAGE_SLOT =
+    0x9aed53e346ef79853612b4c863c4eb308de8a5e84e0fd7d00dad88eb5ff1ea00;
+
+  struct Layout {
+    StakingRewards.Layout staking;
```

Comment:
Can you verify if having a nested struct inside another nested struct won't affect our diamond storage in case we have to add a new variable to StakingRewards.Layout or Treasure and Deposit structs

Reply:
Structs in `StakingRewards` are meant to be immutable and minimal to support staking. Additional variables will be added sequentially to `RewardsDistributionStorage. Layout`.

Reply:
If someone accidentally adds a member to the end of StakingRewards it will be bad, resulting in all old sotrage being sliced.

Is there way to freeze this so that won't happen?

Reply:
I will add natspec comments and a test to ensure `StakingRewards` structs have fixed lengths.

There's also [OpenZeppelin Foundry Upgrades](https://github.com/OpenZeppelin/openzeppelin-foundry-upgrades) but it doesn't support diamond.